pkg/nfd-master/nfd-master.go

```diff
@@ -457,6 +462,31 @@ func (m *nfdMaster) filterFeatureLabels(labels Labels) (Labels, ExtendedResource
 	return outLabels, extendedResources
 }
 
+// Filter annotations by namespace. i.e. adds the possibly missing default namespace for annotations
+// arriving through the gRPC API.
+func (m *nfdMaster) filterFeatureAnnotations(annotations map[string]string) map[string]string {
+	outAnnotations := make(map[string]string)
+
+	for annotation, value := range annotations {
+		// Add possibly missing default ns
+		annotation := addNs(annotation, nfdv1alpha1.FeatureAnnotationNs)
+
+		ns, _ := splitNs(annotation)
+
+		// Check annotation namespace, filter out if ns is not whitelisted
+		if ns != nfdv1alpha1.FeatureAnnotationNs && !strings.HasSuffix(ns, nfdv1alpha1.FeatureAnnotationSubNsSuffix) {
+			// If the namespace is denied, annotation will be ignored
+			if ns == defaultDenyNs || ns == defaultDenySubNs || ns == nfdv1alpha1.AnnotationNs {
```

Comment:
For the subns we want to use strings.HasSuffix

```suggestion
			if ns == defaultDenyNs || ns == strings.HasSuffix(ns, defaultDenySubNs) || ns == nfdv1alpha1.AnnotationNs {
```